engine/function/src/templates/Basic.ftl

```diff
@@ -983,9 +991,11 @@ public class Basic {
     static public ${pt.primitive}[] replaceIfNull(${pt.vector} values, ${pt.primitive} replacement) {
         final int n = values.intSize("replaceIfNull");
         ${pt.primitive}[] result = new ${pt.primitive}[n];
+        int i = 0;
 
-        for (int i = 0; i < n; i++) {
-            result[i] = replaceIfNull(values.get(i), replacement);
+        for (final ${pt.primitive} v : values) {
```

Comment:
More evidence that I did a bad job writing up the issue. This isn't the right pattern to follow, now that we're into the primitives. This approach will box every value (correctly, but we don't want to do it) and then force the language to unbox naively (which will NPE if we boxed to `null`). You'll need to use the iterator explicitly, and invoke `nextChar`, etc to get the primitives, or use `forEachRemaining` with the appropriate primitive consumer.

The `char` version should then look like:
```
try (final CloseablePrimitiveIteratorOfChar iterator = values.iterator()) {
    while (iterator.hasNext()) {
        is (!isNull(iterator.nextChar())) {
            count++;
        }
    }
}
```

This applies more or less to all of the remaining methods you've updated.